### PR TITLE
stm32h5 device supports GPdma

### PR DESCRIPTION
**Pull request by FRASTM**
Add the support of the GP DMA  to the stm32H5 serie
Two GPDMA instances are defined with 8 channels and 140 request
The type of DMA for the stm32h5 serie is similar to the stm32u5.

Signed-off-by: Francois Ramu [francois.ramu@st.com](mailto:francois.ramu@st.com)

### CODE REVIEW

**Comment by FRASTM** [comment]
Running the tests/drivers/dma/chan_blen_transfer on the stm32h573i disco:
```
TESTSUITE dma_m2m succeeded                                                           
                                                                                      
------ TESTSUITE SUMMARY START ------                                                 
                                                                                      
SUITE PASS - 100.00% [dma_m2m]: pass = 4, fail = 0, skip = 0, total = 4 duration = 8.s
 - PASS - [dma_m2m.test_test_dma0_m2m_chan0_burst16] duration = 2.014 seconds         
 - PASS - [dma_m2m.test_test_dma0_m2m_chan0_burst8] duration = 2.014 seconds          
 - PASS - [dma_m2m.test_test_dma0_m2m_chan1_burst16] duration = 2.014 seconds         
 - PASS - [dma_m2m.test_test_dma0_m2m_chan1_burst8] duration = 2.014 seconds          
                                                                                      
------ TESTSUITE SUMMARY END ------      
```

File: drivers/dma/dma_stm32u5.c [hunk 1]
```diff
@@ -24,9 +24,13 @@ LOG_MODULE_REGISTER(dma_stm32, CONFIG_DMA_LOG_LEVEL);
 
 #define DT_DRV_COMPAT st_stm32u5_dma
 
-/* STM32U5 soc has only one GPDMA instance of 15 channels */
+/* STM32U5 soc has only one GPDMA instance of 16 channels */
```
Comment (erwango) [inline drivers/dma/dma_stm32u5.c]: Question on the comment: Is it the GPDMA IP that can have up to 16 channels, or the U5 IP ?

**Review by erwango — DISMISSED** [review]
One minor comment.
Not blocking as the code is fixed in next commit (which is ok if we consider 2 commits are dealing with different stuff)

**Review by gautierg-st — DISMISSED** [review]

File: drivers/dma/dma_stm32u5.c [hunk 2]
```diff
@@ -24,9 +24,13 @@ LOG_MODULE_REGISTER(dma_stm32, CONFIG_DMA_LOG_LEVEL);
...
 #if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
-#define DMA_STM32_0_STREAM_COUNT 16
+#define DMA_STM32_0_STREAM_COUNT DT_INST_PROP(0, dma_channels)
+#endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
+#define DMA_STM32_0_STREAM_COUNT DT_INST_PROP(0, dma_channels)
 #endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
```
[the remark this answers is not in the document]
Comment (FRASTM) [reply drivers/dma/dma_stm32u5.c]: yes, removed in the other commit.
2 commits placed on the stm32u5_dma, could still be made of one.

**Comment by FRASTM** [comment]
Merge two commits for the driver
add the _zephyr/dt-bindings/dma/stm32_dma.h_ to the device DTS

**Review by gautierg-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**